src/data/utils.rs

```diff
@@ -40,3 +41,21 @@ pub fn get_cpu_info() -> Result<CpuInfo> {
     }
     Ok(cpu_info)
 }
+
+pub fn notargz_file_name(pbuf: PathBuf) -> Result<String> {
+    if pbuf.file_name().is_none() {
+        return Ok(String::new());
+    }
+    let mut file_name = pbuf.file_name().unwrap().to_str().unwrap().to_string();
+    if file_name.ends_with(".tar.gz") {
+        file_name = file_name.strip_suffix(".tar.gz").unwrap().to_string();
+    }
+    Ok(file_name)
```

Comment:
Can't this delegate to `notargz_string_name` here?

Reply:
Yes. Will change it.